coding-standards-and-naming.rst

```diff
@@ -193,135 +83,60 @@ Bad class names:
--  ``mysql_xdevapi``: Uses ``mysql_xdevapi`` namespace, e.g.
-   ``mysql_xdevapi\Collection``.
--  ``parallel``: Uses ``parallel`` namespace, e.g. ``parallel\Runtime``.
--  ``parle``: Uses ``Parle`` namespace, e.g. ``Parle\Lexer``.
--  ``pcov``: Uses ``pcov`` namespace, e.g. ``pcov\start()``.
--  ``pq``: Uses ``pq`` namespace, e.g. ``pq\Connection``.
--  ``rdkafka``: Uses ``RdKafka`` namespace, e.g. ``RdKafka\Producer``. Also uses
-   ``RdKafka`` itself, and a handful of ``rd_kafka_*()`` functions.
--  ``xlswriter``: Uses ``Vtiful\Kernel`` namespace, e.g.
-   ``Vtiful\Kernel\Excel``.
--  ``yaf``: Uses ``Yaf`` namespace, e.g. ``Yaf\Application``. Also supports
-   aliases in the global namespace, e.g. ``Yaf_Application``.
--  ``zstd``: Uses ``Zstd`` namespace, e.g. ``Zstd\compress()``. However, it also
-   declares ``zstd_*()`` functions in the global namespace.
-
-It is notable that with the exception of ``xlswriter``, none of these extensions
-make use of a vendor namespace. They all use the package/extension name as the
-top-level namespace. Some extensions additionally have a global class that
-matches the extension name, e.g. the ffi extension uses both ``FFI`` and
-``FFI\CType``.
-
-Proposal
-========
-
-This RFC proposes to explicitly allow and encourage the use of namespaces for
-bundled PHP extensions, subject to the guidelines laid out in the following:
-
--  Extensions should not use a vendor namespace.
--  The top-level namespace should match the extension name (apart from casing).
--  Namespace names should follow ``CamelCase``.
--  All symbols defined in the extension should be part of the extension's
+-  Required extensions (including ``Core`` and ``standard``). These extensions
+   are always present, and PHP cannot be built without them.
+
+-  Bundled extensions (including ``ctype`` and ``mbstring``). These extensions
+   are part of the php-src distribution, but PHP can be built without them.
+   Bundled extensions can be either enabled or disabled by default.
+
+-  3rd-party extensions (including ``apcu`` and ``igbinary``). These extensions
+   are not part of the php-src distribution, and either available through PECL,
+   or on GitHub.
+
+Extensions may move between these three categories over time. ``hash`` and
+``json`` moved from "bundled" to "required", in PHP 7.4 and 8.0 respectively.
+``sodium`` and ``ffi`` moved from 3rd-party to bundled. ``xmlrpc`` and ``wddx``
+moved from bundled to 3rd-party.
+
+Core, Standard, Spl
+===================
+
+Symbols MUST NOT be namespaced under the ``Core``, ``Standard`` or ``Spl``
+namespaces. Instead, these extensions should be considered as a collection of
+different components (``str_*``, ``password_*``), and SHOULD be namespaced
+according to these.
+
+For example, ``str_contains()`` could become ``Str\contains()``, ``fopen()``
+could become ``File\open()``, and ``password_hash()`` could become
+``Password\hash()``.
```

Comment:
I know this comes from the old document, but I find it confusing to lead with these examples, when the document later states `str_` and `array_` as examples that SHOULD NOT use the new namespace naming scheme, but stay consistent with existing functions.

Reply:
I think I've more clearly explained this now.